apigee_edge.module

```diff
@@ -962,6 +962,12 @@ function template_preprocess_app_credential(array &$variables) {
  * it has been in Edge.
  */
 function apigee_edge_user_presave(UserInterface $account) {
+  // If the developer-user synchronization is in progress, then saving
+  // developers while saving Drupal user should be avoided.
+  if (_apigee_edge_get_sync_in_progress()) {
```

Comment:
If the developer sync is in progress and we don't do run this logic, it would mean that the user would not be created in Edge when registering an account in the portal.  This would cause the user to see errors until the developer is synchronized.  Wouldn't we need to add a batch of create user requests to run after sync is done?  This may still cause odd errors if the user is logged in immediately after login and the sync takes a long time.

Reply:
PHP does not preserve the application state between requests, so a static variable lives only through one request. (see `_apigee_edge_set_sync_in_progress()` and `_apigee_edge_is_sync_in_progress()` functions)